src/bevent.c

```diff
@@ -506,8 +506,10 @@ int odict_encode_bevent(struct odict *od, struct bevent *event)
 		if (hdr)
 			err = odict_pl_add(od, "contact", &hdr->val);
 
-		err |= odict_pl_add(od, "display", &msg->from.dname);
-		err |= re_sdprintf(&buf, "%H", uri_encode, &msg->from.uri);
+		if (pl_isset(&msg->from.dname))
+			odict_pl_add(od, "display", &msg->from.dname);
```

Comment:
perhaps check return value?

Reply:
yes, with the check if `dname` is set the only err that might occur is `ENOMEM`. Commit follows